topology/probes/k8s/service.go

```diff
@@ -78,19 +134,46 @@ func (p *serviceProbe) OnDelete(obj interface{}) {
 		p.graph.Lock()
 		defer p.graph.Unlock()
 
-		if nsNode := p.graph.GetNode(serviceUID(srv)); nsNode != nil {
-			p.graph.DelNode(nsNode)
+		if srvNode := p.graph.GetNode(serviceUID(srv)); srvNode != nil {
+			p.graph.DelNode(srvNode)
 			logging.GetLogger().Debugf("Deleted %s", dumpService(srv))
 		}
 	}
 }
 
+func (p *serviceProbe) onNodeUpdated(podNode *graph.Node) {
```

Comment:
you should remove the link to the pod when a node is Deleted : OnNodeDeleted(n *Node)

Reply:
But deletion is handled anyway: https://github.com/skydive-project/skydive/blob/master/topology/graph/graph.go#L1298

Reply:
yes, I forgot that